Assets/Scripts/NPC/CeilingDetection/CeilingRayCastDetection.cs

```diff
@@ -0,0 +1,28 @@
+using UnityEngine;
+
+public class CeilingRayCastDetection : MonoBehaviour
+{
+    [SerializeField] private Transform raycastOrigin;
+    [SerializeField] private float raycastLength = 2f;
+    [SerializeField] private LayerMask ceilingLayer;
+    [SerializeField] private Animator animator;
+
+    private bool isTouchingCeiling = false;
+
+    public bool IsTouchingCeiling => isTouchingCeiling;
+
+    void Update()
+    {
+        RaycastHit hit;
+        if (Physics.Raycast(raycastOrigin.position, Vector3.up, out hit, raycastLength, ceilingLayer))
```

Comment:
```suggestion
        if (Physics.Raycast(transform.position, Vector3.up, out hit, raycastLength, ceilingLayer))
```
De origin is gewoon het pivot point van de gameobject waar dit script op staat, AKA de pivot point van het monster.